src/extension.ts

```diff
@@ -1,96 +1,108 @@
-          const intervalRef = setInterval(() => {
-            if (previewServer.isServerRunning()) {
-              clearInterval(intervalRef);
-              resolve();
-              if (!statusBarItem) {
-                statusBarItem = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right, 10);
-                statusBarItem.command = 'asyncapi.stop';
-                statusBarItem.text = 'AsyncAPI Preview';
-                statusBarItem.tooltip = 'Stop AsyncAPI Preview Server';
-                statusBarItem.show();
-                context.subscriptions.push(statusBarItem);
-              }
-            }
-          }, 1000);
-        });
-      }
-    );
+  vscode.window.onDidChangeActiveTextEditor(e => {
+    if (e?.document) {
+      setAsyncAPIPreviewContext(e.document);
+    }
   });
-  vscode.workspace.onDidChangeTextDocument((e: vscode.TextDocumentChangeEvent) => {
-    if (e.document === vscode.window.activeTextEditor.document) {
-      previewServer.update(e.document.fileName, e.document.getText());
+
+  vscode.workspace.onDidSaveTextDocument(document => {
+    if (openAsyncapiFiles[document.uri.fsPath]) {
+      console.log('Reloading asyncapi file', document.uri.fsPath);
+      openAsyncAPI(context, document.uri);
     }
   });
+
+  let disposable = vscode.commands.registerCommand('asyncapi.preview', (uri: vscode.Uri) => {
+    console.log('Opening asyncapi file', uri.fsPath);
+    openAsyncAPI(context, uri);
+  });
+
   context.subscriptions.push(disposable);
-  context.subscriptions.push(
-    vscode.commands.registerCommand('asyncapi.stop', () => {
-      previewServer.stop();
-      if (statusBarItem) {
-        statusBarItem.hide();
-        statusBarItem.dispose();
-        statusBarItem = null;
-      }
-    })
-  );
 }
 
-export function deactivate() {
-  if (previewServer) previewServer.stop();
+function isAsyncAPIFile(text: string) {
+  return text.includes('asyncapi:');
 }
+
+function openAsyncAPI(context: vscode.ExtensionContext, uri: vscode.Uri) {
+  const panel: vscode.WebviewPanel =
+    openAsyncapiFiles[uri.fsPath] ||
+    vscode.window.createWebviewPanel('asyncapi-preview', '', vscode.ViewColumn.Two, {
+      enableScripts: true,
+      retainContextWhenHidden: true,
+      localResourceRoots: [
+        vscode.Uri.file(path.dirname(uri.fsPath)),
+        vscode.Uri.joinPath(context.extensionUri, 'dist/node_modules/@asyncapi/react-component/browser/standalone'),
+        vscode.Uri.joinPath(context.extensionUri, 'dist/node_modules/@asyncapi/react-component/styles'),
+      ],
+    });
+  panel.title = path.basename(uri.fsPath);
+  panel.webview.html = getWebviewContent(context, panel.webview, uri);
+
+  panel.onDidDispose(() => {
+    delete openAsyncapiFiles[uri.fsPath];
+  });
+  openAsyncapiFiles[uri.fsPath] = panel;
+}
+
+function getWebviewContent(context: vscode.ExtensionContext, webview: vscode.Webview, asyncapiFile: vscode.Uri) {
+  const asyncapiWebComponentJs = webview.asWebviewUri(
```

Comment:
I don't think you are using a Web Component but rather a Standalone bundle -> https://github.com/asyncapi/asyncapi-react/blob/next/docs/usage/standalone-bundle.md

Reply:
it's just the name of a variable, but updated